hack/generate-fakes.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -euo pipefail
+
+[ ! -d "vendor" ] && echo "$0 requires vendor/ folder, run 'go mod vendor'"
+
+counterfeiter -o pkg/controller/fakes/manager.go vendor/sigs.k8s.io/controller-runtime/pkg/manager Manager
```

Comment:
Could you add some dev docs around using ginkgo / counterfeiter for this project in HACK.md ?

Reply:
definitely.